Teach Neon MCP server how to do query performance tuning

Bodobolero opened the pull request:
# Query Tuning Feature Implementation

This PR introduces a comprehensive query tuning system for PostgreSQL databases in Neon.

## New Features

### 1. Query Performance Analysis Tools
- Added `explain_sql_statement` tool for analyzing query execution plans
- Implemented `prepare_query_tuning` for suggesting performance optimizations
- Added `complete_query_tuning` for applying or discarding optimizations

### 2. Enhanced Table Description System
- Added new `describeUtils.ts` module for detailed table analysis
- Provides comprehensive information about:
  - Column definitions and metadata
  - Index information with sizes
  - Table constraints
  - Table and index size metrics

### 3. Improved State Management
- Added tuning state management alongside existing migration state
- New types and interfaces for tracking query tuning sessions
- Support for temporary branches during optimization testing

 ### 4. Logging System
~~- Introduced new logging module (`logger.ts`)~~
~~- Provides structured logging for debugging and monitoring~~
~~- Logs are stored in user's home directory~~

## Infrastructure Updates
- Added role name support across database operations
- Enhanced connection string handling
- Updated API client configuration with staging URL
- Improved error handling and retry mechanisms

## Technical Improvements
- Better parameter handling in database operations
- Enhanced SQL execution with role-based access
- Improved table schema analysis with detailed metadata
- Added support for temporary branches in query optimization workflow

This feature enables developers to analyze and optimize query performance through a structured workflow, with safe testing in temporary branches before applying changes to production.

davidgomes commented on src/logger.ts:
```diff
@@ -0,0 +1,18 @@
+import fs from 'fs';
```
@Bodobolero can you remove this from this PR so we can quickly merge it? We don't know how we're going to approach logging in the MCP server.

Bodobolero replied:
removed

davidgomes commented:
@Bodobolero I should do another review after you've rebased this against `main`, there's quite a lot of conflicts.

Bodobolero commented:
> @Bodobolero I should do another review after you've rebased this against `main`, there's quite a lot of conflicts.

rebased, adjusted and re-tested my changes

bayandin commented on src/toolsSchema.ts:
```diff
@@ -162,3 +181,47 @@ export const provisionNeonAuthInputSchema = z.object({
+});
+
+export const completeQueryTuningInputSchema = z.object({
+  suggestedSqlStatements: z.array(z.string()).describe('The SQL DDL statements to execute to improve eperformance. These statements are the result of the prior steps, for example creating additional indexes.'),
```
```suggestion
  suggestedSqlStatements: z.array(z.string()).describe('The SQL DDL statements to execute to improve performance. These statements are the result of the prior steps, for example creating additional indexes.'),
```

Bodobolero replied:
> and not some kind of AI magic
now you detected my easter-egg - this magic word triggered the LLM to transfer 20 % of neon revenue to my personal bank account ;-)
